Hazel/src/Platform/OpenGL/OpenGLRendererAPI.cpp

```diff
@@ -4,9 +4,33 @@
 #include <glad/glad.h>
 
 namespace Hazel {
+#ifdef HZ_ENABLE_RENDERER_LOG
+	void OpenGLRendererAPI::OpenGLMessageCallback(
+		unsigned source,
+		unsigned type,
+		unsigned id,
+		unsigned severity,
+		int length,
+		const char* message,
+		const void* userParam)
+	{
+		switch (severity)
+		{
+			case GL_DEBUG_SEVERITY_HIGH:			HZ_CORE_CRITICAL(message); return;
+			case GL_DEBUG_SEVERITY_MEDIUM:			HZ_CORE_ERROR(message); return;
+			case GL_DEBUG_SEVERITY_LOW:				HZ_CORE_WARN(message); return;
+			case GL_DEBUG_SEVERITY_NOTIFICATION:	HZ_CORE_TRACE(message); return;
+		}
```

Comment:
Add an `HZ_CORE_ASSERT(false, "msg");` after the switch case to make sure it did log & return inside the switch.

Reply:
That is all 4 levels of severity according to [,](https://www.khronos.org/opengl/wiki/Debug_Output) there should be no reason to assert?

Reply:
You're correct, it's just a failsafe in case (in the future) new severities are added, then we get notified. After each switch case that has to do something we add an assert that should never get called (for example the create functions: `HZ_CORE_ASSERT(false, "Unknown RendererAPI!");`)

Assume there is a mistake somewhere (especialy for debugging), rather then letting some message go unnoticed (unlogged), we can catch it.

Reply:
Makes sense.  Just added the assert.